packages/eslint-plugin/src/rules/consistent-type-imports.ts

```diff
@@ -142,6 +142,7 @@ export default createRule<Options, MessageIds>({
               } else {
                 if (
                   !sourceImports.valueOnlyNamedImport &&
+                  node.specifiers.length &&
```

Comment:
> any way to distinguish between these two cases without looking for the braces explicitly

If you do end up needing to distinguish the two cases, `context` has some APIs around tokens.

```plaintext
> context.sourceCode.getText(node)
'import {} from 'foo';'

> context.getTokenBefore(node.source)
{type: 'Identifier', value: 'from', range: Array(2), loc: {…}, start: <accessor>, …}

> context.getTokens(node)
(6) [{…}, {…}, {…}, {…}, {…}, {…}]
0: {type: 'Keyword', value: 'import', range: Array(2), loc: {…}, start: <accessor>, …}
1: {type: 'Punctuator', value: '{', range: Array(2), loc: {…}, start: <accessor>, …}
2: {type: 'Punctuator', value: '}', range: Array(2), loc: {…}, start: <accessor>, …}
3: {type: 'Identifier', value: 'from', range: Array(2), loc: {…}, start: <accessor>, …}
4: {type: 'String', value: ''foo'', range: Array(2), loc: {…}, start: <accessor>, …}
5: {type: 'Punctuator', value: ';', range: Array(2), loc: {…}, start: <accessor>, …}
```

```plaintext
> context.sourceCode.getText(node)
'import 'foo';'

> context.getTokenBefore(node.source)
{type: 'Keyword', value: 'import', range: Array(2), loc: {…}, start: <accessor>, …}

> context.getTokens(node)
(3) [{…}, {…}, {…}]
0: {type: 'Keyword', value: 'import', range: Array(2), loc: {…}, start: <accessor>, …}
1: {type: 'String', value: ''foo'', range: Array(2), loc: {…}, start: <accessor>, …}
2: {type: 'Punctuator', value: ';', range: Array(2), loc: {…}, start: <accessor>, …}
```

Hope this helps!

Reply:
Ah this is helpful. As outlined above, I've opted to keep the current implementation, but this is good context for future work.